specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,568 @@
+
+### Masking
+
+Individual _artifact verifiers_ and _trust anchors_ may be masked using a symlink to `/dev/null`, independent of [technology].
+This constitutes the only allowed exception to the rule of not symlinking to files or directories external to the load paths.
+
+These symlinks are only expected to be used in writable and persistent load paths (that is, only below `/etc/voa/` when in system mode, or below`$XDG_DATA_HOME/voa/` when in user mode).
+Masked verifiers are not considered when retrieving from any of the VOA [load paths].
+
+Masking _artifact verifiers_ and _trust anchors_ may be desirable in situations where a verifier is considered compromised, but a revocation for it can not be provided to users in time.
+VOA offers this mechanism as an alternative to revocation.
+
+### Load logic
+
+Load logic depends on the [technology] in use.
+It is the mechanism by which relevant verifier information from all load paths is retrieved, while considering the [os], [purpose], [context] and [technology] request of a calling application.
+
+For some technologies, the first suitable _signature verifier_ found in the list of [load paths] is used, effectively [overriding] files with the same name in directories lower in the list.
+
+However, in other technologies, more complex [merging] logic may apply, that effectively combines information from different representations of one verifier.
+
+Additionally, the concept of [masking] applies for all technologies.
+
+#### Overriding
+
+In technologies that use overriding, verifier files shadow other variants of the same verifier based on filename equality.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver1.pub` overrides `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`.
+
+Overriding depends on file name equality and directory structure.
+
+For example, `/etc/voa/fedora/packages/default/ssh/buildserver2.pub` does not override `/usr/share/voa/fedora/package/default/ssh/buildserver1.pub`, but instead adds an additional _signature verifier_.
+
+Great care has to be taken to ensure the naming scheme has the desired overriding behavior (e.g. to make sure that a revoked version of an SSH public key overrides a non-revoked version of the same SSH public key found in a different load path).
+
+#### Merging
+
+Some technologies use a business logic that "merges" different versions of the same _signature verifier_ (instead of applying overriding semantics).
+The purpose is to use the latest available information about each verifier, combining the information from all sources.
+
+When merging, information about a verifier found anywhere in the current _verifier lookup_ must be applied for all uses of a verifier (e.g. for revocation status).
+
+Wherever it is applicable and feasible, VOA technology backend implementations should implement "merging" to obtain a unified view of the latest information for each verifier.
```

Comment:
What would be the outcome if a OpenPGP key has expired in one signature verifier entry, and another version has been modified with a new expiry date. I am guessing the new expiry date takes prioerity? Or would it be counted as revoked/expired still?